src/kubectlUtils.ts

```diff
@@ -0,0 +1,107 @@
+import * as vscode from "vscode";
+import { Kubectl } from "./kubectl";
+
+export type Cluster = {
+    name: string;
+    context: string;
+    user: string;
+    active: boolean;
+};
+
+export type Namespace = {
+    name: string;
+    active: boolean;
+};
+
+async function getKubeconfig(kubectl: Kubectl): Promise<any> {
+    const shellResult = await kubectl.invokeAsync("config view -o json");
+    if (shellResult.code !== 0) {
+        vscode.window.showErrorMessage(shellResult.stderr);
+        return null;
+    }
+    return JSON.parse(shellResult.stdout);
+}
+
+export async function getClusters(kubectl: Kubectl): Promise<Cluster[]> {
+    const kubectlConfig = await getKubeconfig(kubectl);
+    if (!kubectlConfig) {
+        return;
+    }
+    const currentContext = kubectlConfig["current-context"];
+    const contexts = kubectlConfig.contexts;
+    return contexts.map((c) => {
+        return {
+            name: c.context.cluster,
+            context: c.name,
+            user: c.context.user,
+            active: c.name === currentContext
+        };
+    });
+}
+
+export async function deleteCluster(kubectl: Kubectl, cluster: Cluster): Promise<boolean> {
+    const deleteClusterResult = await kubectl.invokeAsyncWithProgress(`config delete-cluster ${cluster.name}`, "Deleting cluster...");
+    if (deleteClusterResult.code !== 0) {
+        await vscode.window.showErrorMessage(`Failed to delete the specified cluster ${cluster.name} from the kubeconfig: ${deleteClusterResult.stderr}`);

```

Comment:
Will this fit in the small VS Code message window?  Might want to look at the "Failed, see output window for why" cop-out that we use in a couple of other places where the size limit was a problem...

Reply:
Good suggestion, let me reference the similar UX in other places.